src/HttpClient/HttpClientFactory.php

```diff
@@ -110,7 +111,7 @@ public function create(Options $options): HttpAsyncClientInterface
      */
     private function resolveClient(Options $options)
     {
-        if (class_exists(SymfonyHttplugClient::class)) {
+        if (ConcreteImplementation::VENDOR_SYMFONY === ConcreteImplementation::HTTPLUG_VENDOR) {
```

Comment:
Using these consts gives more control over which implementation should be used (the consts are generated by the plugin xor computed once at autoload-time when the plugin is disabled)